monai/transforms/spatial/dictionary.py

```diff
@@ -646,6 +805,9 @@ def __init__(
                 See also: https://pytorch.org/docs/stable/generated/torch.nn.functional.grid_sample.html
                 It also can be a sequence of string, each element corresponds to a key in ``keys``.
             device: device on which the tensor will be allocated.
+            dtype: data type for resampling computation. Defaults to ``np.float32``.
```

Comment:
May I know why the `dtype` of resampling computation is `float32` and not consistent with `SpatialResampled`?

Thanks.

Reply:
the primary use case is random data augmentation, float32 is good enough with the speed and memory and compatibility